### PR TITLE
Upgrade pip for cache installation

### PR DESCRIPTION
**Pull request by OrkoHunter**
Travis-CI still uses 6.x version of pip. However, pip 7.0.0 and later cache installed packages automatically in the form of wheels. Subsequent installs are therefore much faster which results in decrease of the total installation time.

### CODE REVIEW

**Comment by hagberg** [comment]
Why does this matter with Travis?  There is no persistent cache between tests right?

**Comment by ysitu** [comment]
@hagberg Caching is possible with the move to Travis' container-based CI environment ([doc](http://docs.travis-ci.com/user/caching/#pip-cache)).

File: README.rst [hunk 1]
```diff
@@ -41,5 +41,4 @@ Distributed with a BSD license; see LICENSE.txt::
    Pieter Swart <swart@lanl.gov>
 
 
-
-
+[Add a test line]
```
Comment (ysitu) [inline README.rst]: This seems unnecessary to me. Travis builds after every commit.

**Comment by OrkoHunter** [comment]
@ysitu That was a test commit. Sorry for that.

@hagberg After this change, Travis will cache the wheels and upload them to [S3](http://docs.travis-ci.com/user/deployment/s3/). So that the **builds on the commits after this change**, would be able to use them.
See [this](https://github.com/nickstenning/travis-pip-cache) for an example of how 2 min 28 sec drastically decreased to 16 sec

**Comment by ysitu** [comment]
If the cached wheels become outdated, will pip update them?

**Comment by OrkoHunter** [comment]
I didn't think about that. Should we create an issue over https://github.com/pypa/pip or https://github.com/travis-ci/travis-web ?

**Comment by ysitu** [comment]
According to the [doc](https://pip.pypa.io/en/latest/reference/pip_install.html#caching), pip updates the cache the same way browsers do. That looks good enough to me.

**Pull request merged**